src/main/scala/zio/codec/CharCodecModule.scala

```diff
@@ -0,0 +1,199 @@
+package zio.codec
+
+import zio.Chunk
+import zio.internal.Stack
+
+trait CharCodecModule extends CodecModule {
+  type Input = Char
+
+  def decoder[A](codec: Codec[A]): Chunk[Input] => Either[DecodeError, (Int, A)] = {
+    val compiled = compileCodec(codec, 0)
+//    println(compiled.zipWithIndex.map { case (o, i) => i.toString.reverse.padTo(3, '0').reverse + ": " + o }.mkString("", "\n", "\n"))
+    interpret(_, compiled).asInstanceOf[Either[DecodeError, (Int, A)]]
+  }
+
+  def encoder[A](codec: Codec[A]): A => Either[EncodeError, Chunk[Input]] = ???
+
+  def printer[A](codec: Codec[A]): List[String] = ???
+
+  private case object NoValue
+
+  private def compileCodec[A](codec: Codec[A], offset: Int): Array[CodecVM] = {
+    import Codec._
+    val VM = CodecVM
+
+    codec match {
+      case Produce(a) =>
+        Array(
+          VM.Push(a.asInstanceOf[AnyRef])
+        )
+
+      case Fail(error) =>
+        Array(
+          VM.Fail(error)
+        )
+
+      case Consume =>
+        Array(
+          VM.Read(None, None)
+        )
+
+      case Map(_, value) =>
+        // todo
+        compileCodec(value(), offset) ++ Array[CodecVM](
+          )
+
+      case Filter(value, in, not) =>
+        val program = compileCodec(value(), offset) ++ Array(
+          VM.Duplicate,
+          VM.CheckSet(in.map(_.asInstanceOf[AnyRef])),
+          VM.Push((!not).asInstanceOf[AnyRef])
+        )
+
+        val lMismatch = offset + program.length + 1
+        val lMatch    = offset + program.length + 3
+
+        program ++ Array(
+          VM.JumpCond(lMatch, lMismatch),
+          VM.Pop,
+          VM.Push(NoValue),
+          VM.Noop // todo: not needed
+        )
+
+      case Zip(left, right) =>
+        val program = compileCodec(left(), offset)
+        program ++ compileCodec(right(), offset + program.length) ++ Array(
+          VM.Construct2(Tuple2.apply)
+        )
+
+      case Alt(_, _) =>
+        // todo
+        ???
```

Comment:
To implement this one, it is necessary for the VM to support a "catch" like construct.

A simple one could look like:

```scala
Try(length: Int, ifFail: Int)
```

where `length` describes the length of the "try" block, and `ifFail` describes the instruction to jump to if there is a failure inside there.

This might not be ideal. You could also look into how the JVM does exceptions, with an exception table (though that's probably too fancy for our needs here).

Reply:
Right now I have a `NoValue` object placed on the stack to indicate that codec didn't produce a value. For example `Rep` is done this way.

Reply:
Ok, worth trying if that's sufficient. The simplest possible thing to get the job done should be preferred.